jax_galsim/moffat.py

```diff
@@ -0,0 +1,444 @@
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+
+import jax.scipy as jsc
+
+
+from jax_galsim.gsobject import GSObject
+from jax_galsim.gsparams import GSParams
+from jax_galsim.core.draw import draw_by_xValue, draw_by_kValue
+
+from jax_galsim.core.bessel import j0
+from jax_galsim.core.integrate import ClenshawCurtisQuad, quad_integral
+
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+from jax.tree_util import register_pytree_node_class
+
+import tensorflow_probability as tfp
+
+
+##JEC 2/8/23
+##first import in the context of new JAX-Galsim repo
+
+
+##JEC 7/2/23
+## Truncated Moffat (start)
+
+
+##JEC 20/1/23
+## Moffat profile implementation
+## Nb: The maxK in case of trunc =0 differs from the GalSim implementation
+##     see https://github.com/GalSim-developers/GalSim/issues/1208
+##     To allow a comparison of GalSim/Jax-GalSim I code also the original GalSim expression
```

Comment:
I understand. In fact the comment 20/1/23 is may be the only one that matter.
```
Nb: The maxK in case of trunc =0 differs from the GalSim implementation
    see hxttps://github.com/GalSim-developers/GalSim/issues/1208
    To allow a comparison of GalSim/Jax-GalSim I code also the original GalSim expression
```